frontend/src/host_orchestrator/orchestrator/userartifacts_test.go

```diff
@@ -414,6 +416,155 @@ func TestStatArtifactSucceeds(t *testing.T) {
 	}
 }
 
+func TestExtractArtifactSucceedsWithZipFormat(t *testing.T) {
+	legacyRootDir := orchtesting.TempDir(t)
+	defer orchtesting.RemoveDir(t, legacyRootDir)
+	rootDir := orchtesting.TempDir(t)
+	defer orchtesting.RemoveDir(t, rootDir)
+	opts := UserArtifactsManagerOpts{LegacyRootDir: legacyRootDir, RootDir: rootDir}
+	uam, err := NewUserArtifactsManagerImpl(opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	data, err := ioutil.ReadFile("test_data/test.zip")
+	if err != nil {
+		t.Fatal(err)
+	}
+	checksum := getSha256Sum(data)
+	chunk := UserArtifactChunk{
+		Name:          "test.zip",
+		OffsetBytes:   0,
+		SizeBytes:     int64(len(data)),
+		FileSizeBytes: int64(len(data)),
+		File:          bytes.NewReader(data),
+	}
+	if err := uam.UpdateArtifact(checksum, chunk); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := uam.ExtractArtifact(checksum); err != nil {
+		t.Fatal(err)
+	}
+	cmd := exec.Command("diff", "-r", "test_data/zip_extracted", filepath.Join(rootDir, fmt.Sprintf("%s_extracted", checksum)))
```

Comment:
There's no need of relying on an external tool like `diff` here. 

Make it simple by compressing only one file and then verify the content of that single file programatically using  "github.com/google/go-cmp/cmp" package.  Delete  the `zip_extracted` and `tar_gz_extracted` directories.

If you want to add more files, verify its existence and content programatically making the test hermetic.